pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -76,32 +76,41 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 		if !isMlnxNicAndInNode(iface.PciAddress, new) {
 			continue
 		}
-		fwCurrent, fwNext, err := getMlnxNicFwData(iface.PciAddress)
+		fwData, err := getMlnxNicFwData(iface.PciAddress)
 		if err != nil {
 			return false, false, err
 		}
-		if fwCurrent.enableSriov != fwNext.enableSriov || fwCurrent.totalVfs != fwNext.totalVfs ||
```

Comment:
Why this is not needed? Does it compare totalVfs between Current and Next somewhere?

Reply:
So I think it better that we first merge the PR with the dual port fix and then @Mmduh-483 wll rebase this PR.

Reply:
@zshi-redhat this check won't be needed anymore, there is 2 checks (link type removed in this PR) and the (sriovTotal and sriovEnable) will be removed in the reset the FW total VFs